### DISCUSSION TITLE
Data in mq container won't persist

### DISCUSSION THREAD

**Discussion by LukaszPa** [ibm-messaging/mq-container#530] [Q&A] [answered]
Hi, 

   I'm building mq container but I have problem with data persist.

DockerFile 
---------------------------------------------------------------------
FROM icr.io/ibm-messaging/mq:latest
ENV LICENSE=accept
ENV MQ_QMGR_NAME=LOCAL
ENV MQSNOAUT=yes
#code with commands to create queues
COPY ["mqScripts/setup-mq.sh", "/opt/mqm/setup-mq.sh"]
---------------------------------------------------------------------
while starting image iIm attaching docker volume ->  myVolume:/mnt/mqm

When container starts I log in to container terminal and start command to create queues. Then I'm sending some dummy massages to new queues. I see that messages are waiting on queue.
  
  After I stop and start mq container all massages are lost. Only new queues are persist.

 What could be the reason ? Please help :)

**Comment by arthurbarr** [comment, marked as answer]
Are you using persistent messages?  Message persistence is a property of the message, so usually in control of the application putting the message.  The application can also defer to administrative settings with the queue property (`DEFPSIST`): https://www.ibm.com/docs/en/ibm-mq/9.3?topic=reference-define-queues

**Reply by LukaszPa** [reply]
Thanks for fast response - it was message issue :) but good to know how to pass this setting to queue :)
